src/compiler/checker.ts

```diff
@@ -35500,10 +35525,12 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             }
 
             if (reportErrors) {
-                const diag = createDiagnosticForNode(node.tagName, Diagnostics.Tag_0_expects_at_least_1_arguments_but_the_JSX_factory_2_provides_at_most_3, entityNameToString(node.tagName), absoluteMinArgCount, entityNameToString(factory), maxParamCount);
-                const tagNameDeclaration = getSymbolAtLocation(node.tagName)?.valueDeclaration;
+                const errorNode = isJsxOpeningFragment(node) ? node : node.tagName;
+                const errorEntityName = isJsxOpeningFragment(node) ? "Fragment" : entityNameToString(node.tagName);
+                const diag = createDiagnosticForNode(errorNode, Diagnostics.Tag_0_expects_at_least_1_arguments_but_the_JSX_factory_2_provides_at_most_3, errorEntityName, absoluteMinArgCount, entityNameToString(factory), maxParamCount);
+                const tagNameDeclaration = isJsxOpeningFragment(node) ? getJSXFragmentType(node).symbol.valueDeclaration : getSymbolAtLocation(node.tagName)?.valueDeclaration;

```

Comment:
I did some testing, and even with proper checking of fragments in the rest of `checkTagNameDoesNotExpectTooManyArguments`, it doesn't seem possible to hit this error message with a fragment, so I reverted this change. Hitting this error message requires attributes to be passed in through the tag (example `<TagName x={/*stuff*/}>`). For just an opening element, if there are no attributes in the tag (`<TagName>`), we won't ever hit this error because we issue missing property or not assignable errors instead. So even if `Fragment` is defined to always take multiple arguments, we won't ever get this error